metadata-ingestion/setup.py

```diff
@@ -259,7 +259,8 @@
 
 delta_lake = {
     *s3_base,
-    "deltalake>=0.6.3, != 0.6.4",
+    "deltalake>=0.6.3, != 0.6.4, < 0.18.0; platform_system == 'Darwin' and platform_machine == 'arm64'",
+    "deltalake>=0.6.3, != 0.6.4; platform_system != 'Darwin' or platform_machine != 'arm64'",
```

Comment:
Seems like 0.18.0 is broken on ARM macs. Can alternately just forbid version 0.18.0 but may run into the same issue on the next version

Reply:
can we also link to the issue https://github.com/delta-io/delta-rs/issues/2577